task/github-create-deployment/0.1/github-create-deployment.yaml

```diff
@@ -0,0 +1,149 @@
+      description: |
+        The GitHub repository full name, i.e: tektoncd/catalog.
+      type: string
+
+    - name: REF
+      description: |
+        The ref to deploy. This can be a branch, tag, or SHA.
+      type: string
+
+    - name: TASK
+      description: |
+        Specifies a task to execute (e.g., deploy or deploy:migrations).
+      default: "deploy"
+      type: string
+
+    - name: AUTO_MERGE
+      description: |
+        Attempts to automatically merge the default branch into the
+        requested ref, if it's behind the default branch.
+      default: "true"
+      type: string
+
+    - name: REQUIRED_CONTEXTS
+      description: |
+        The status contexts to verify against commit status checks. To
+        verify all unique contexts before creating a deployment, pass
+        in "*". To bypass checking entirely, pass an empty string.
+      default: "*"
+
+    - name: PAYLOAD
+      description: |
+        JSON payload with extra information about the deployment.
+      default: ""
+      type: string
+
+    - name: ENVIRONMENT
+      description: |
+        Name for the target deployment environment (e.g., production, staging).
+      type: string
+
+    - name: DESCRIPTION
+      description: |
+        Short description of the deployment.
+      default: ""
+      type: string
+
+    - name: GITHUB_TOKEN_SECRET_NAME
+      description: |
+        The name of the Kubernetes Secret that contains the GitHub token.
+      type: string
+      default: github
+
+    - name: GITHUB_TOKEN_SECRET_KEY
+      description: |
+        The key within the Kubernetes Secret that contains the GitHub token.
+      type: string
+      default: token
+
+  steps:
+    - name: create-deployment
+      env:
+        - name: GITHUBTOKEN
+          valueFrom:
+            secretKeyRef:
+              name: $(params.GITHUB_TOKEN_SECRET_NAME)
+              key: $(params.GITHUB_TOKEN_SECRET_KEY)
+
+      image: registry.access.redhat.com/ubi8/ubi:8.2
+      script: |
+        #!/usr/libexec/platform-python
+        import http.client
+        import json
+        import os
+        import sys
+
+        deployment_url = "/repos/$(params.REPO_FULL_NAME)/deployments"
+
+        if "$(params.REQUIRED_CONTEXTS)" == "*":
+            required_contexts = None
+        elif "$(params.REQUIRED_CONTEXTS)" == "":
+            required_contexts = []
+        else:
+            required_contexts = "$(params.REQUIRED_CONTEXTS)".split(",")
+
+        data = {
+          "ref": "$(params.REF)",
+          "task": "$(params.TASK)",
+          "auto_merge": json.loads("$(params.AUTO_MERGE)"),
+          "required_contexts": required_contexts,
+          "payload": json.loads("$(params.PAYLOAD)") if "$(params.PAYLOAD)" else None,
+          "environment": "$(params.ENVIRONMENT)",
+          "description": "$(params.DESCRIPTION)" if "$(params.DESCRIPTION)" else None,
+          }
+        data = { k: v for k, v in data.items() if v is not None }
```

Comment:
are you sure it's is not None you want ? you want to check if the string is empty isnt it ? 

I am actually not sure what this line does ? have you copied from somewhere else ?

Reply:
So what this line does is prune keys from our data payload if its value is None. The reason this is needed is because GitHub will reject a payload with empty string values, eg. `{ description: "" }`.

This line essentially translates into
```python
for k, v in data.items():
    if v is not None:
        data[k] = v
```

Reply:
sounds good 👍

Reply:
So it seems my earlier statement was incorrect 🤦 (been quite awhile since I did this so context has been lost)

Turns out we don't need to prune empty string values from the payload - my previous implementation was to ensure we can do `json.loads()` correctly on `$(params.PAYLOAD)`, as `json.loads("")` raises an error. I've now modified the script to reflect this and it works well from testing.

Previously:
- Set to `None` if empty string, else `json.loads()`
- Prune None values

Currently:
- If not empty string, do `json.loads()`
- Else leave as empty string

Have also added a comment to explain this: https://github.com/tektoncd/catalog/pull/258/files#diff-ba3bbc0b3249e2e2fce3c0366a4b451dd81de4b117cbd95b9ccb64a01a263659R108

Sorry for the confusion and big roundabout.. 🙈 Hope this makes more sense?